types/types.go

```diff
@@ -92,24 +100,59 @@ type Marker interface {
 	Inhibited(model.Fingerprint) ([]string, bool)
 }
 
-// NewMarker returns an instance of a Marker implementation.
-func NewMarker(r prometheus.Registerer) Marker {
-	m := &memMarker{
-		m: map[model.Fingerprint]*AlertStatus{},
-	}
+type GroupMarker interface {
+	// Muted returns true if the group is muted, otherwise false. If the group
+	// is muted then it also returns the names of the time intervals that muted
+	// it.
+	Muted(groupKey string) ([]string, bool)
```

Comment:
In the original version of `GroupMarker` I had:

```go
type GroupMarker interface {
	// Muted returns true if the group is muted, otherwise false. If the group
	// is muted then it also returns the names of the time intervals that muted
	// it.
	Muted(groupKey string, fingerprint model.Fingerprint) ([]string, bool)
	...
}
```

but then I realized we didn't need to store the fingerprint at all.

The reason here is that active and mute timings work against routes. That means either all alerts in an aggregation group are suppressed because of time intervals, or none of them are. It is not possible to have two alerts A and B in the same aggregation group, where A is muted from a mute time interval and B is not.

Reply:
We have found an issue where because `groupKey` is not guaranteed to be unique, it's possible for two (or more) different groups to have the same `groupKey`.  In such cases, where one group is muted and another group is not, both will be marked as muted.

Reply:
The following YAML shows a configuration containing two routes with the same `groupKey`. This happens because the `groupKey` is calculated using:

1. The matchers in the route
2. The `group_by` labels (default is empty)

And both routes have the same matchers and `group_by` labels.

```yml
receivers:
  - name: test1
  - name: test2
route:
  receiver: test1
  routes:
    - receiver: test1
      matchers:
        - foo=bar
    - receiver: test2
      matchers:
        - foo=bar
      mute_time_intervals:
        - name: weekends
```

The reason a user might have such a configuration is to mute notifications on the weekends, but still send webhooks to an issue tracker.